### ISSUE TITLE
[Bug]: Hidden folders still show up if they contain non hidden subfolders

### ISSUE THREAD

**Issue by hakaserver** [BlafKing/sd-civitai-browser-plus#111]
### Describe the bug.

The option recently implemented hides folders beggining with "." correctly, only if they are the last folder of the path. However, if these folders contain other folders (without names beggining with "."), the subfolders of these hidden folders still show up in the "Subfolder" list.

Example of folder tree:

.~\\.hidden_folder~&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<- this folder is correctly hidden
.\\.hidden_folder\folder1\ &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<- however this folder still shows up
.~\\.hidden_folder\folder1\\.git~&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<- correctly hidden
.\\.hidden_folder\folder1\\.git\folder2&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<- even if there are multiple hidden folders (like a .git folder) it shows up
.\normal_folder\ &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<- normal folder shows up
.~\normal_folder\\.hidden_folder2~&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<- correctly hidden
.\normal_folder\\.hidden_folder2\folder3&nbsp;&nbsp;&nbsp;<- hidden folders inside normal folders should also be hidden if sub-subfolders exist
.\normal_folder\folder4&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<- normal folder shows up

### Steps to reproduce the problem.

1. Enable option to hide folders beggining with "."
2. Restart UI
3. Open the "Subfolder" list when downloading a model
4. Sublist still show folder structures inside of hidden folders, despite not showing the hidden folder itself

### Expected behavior

The expected behaviour would be to hide the folder beggining with "." but also every other folder contained in such hidden folder.
In other words, paths containing a folder beggining with "." should never appear on the list, even if the path ends in a normal folder.
In the example showed above, the only folders that should appear in the list are:

.\normal_folder\
.\normal_folder\folder4\

### System info

* Extension version: main/ef12ec84
* OS: Win
* SD-WebUI version: 1.6.0
* Python: 3.10.6

### Console logs

```Shell
not applicable
```

**Comment by BlafKing** [comment]
Thanks for the report!

I didn't take into account that it would show any sub folders 😅 
This issue should now be fixed in [v2.0.1](https://github.com/BlafKing/sd-civitai-browser-plus/commit/95f6d8e45f6bfd45feb4010625844500685eadad)! :)
